Link avatar image

Envl opened the pull request:
Things work now, but the with..... Files look quite counter-intuition to me.
Since they just attach some object to a component, there might be other clearer solutions.
I might look at that if there will be time after submitting the mid-project.

Envl committed:
link the address of portrait pictaure after uploading
deleted code storing and deleting authUser using localStorage because they seems did nothing at all

Envl committed:
make default portrait image

Envl committed:
bug fix

Envl committed:
delete logs

Envl committed:
resolve conflict

bt-s committed:
Remove obsolete code

bt-s committed:
Merge branch 'link-avatar-image' of github.com:bt-s/ipdw-project into link-avatar-image

bt-s commented on src/components/Account/Avatar.js:
```diff
@@ -26,7 +26,18 @@ const Avatar = props => {
+      .updateProfile({
+        photoURL: url,
+      })
+      .then(function() {
```
Maybe write some console output here; now the then and catch of this Promise do nothing.

bt-s committed:
Fix typo

bt-s committed:
Fix bug introduced after merging master into feature branch

It's bad practice to merge master into the feature branch, since that way you
 can really easily introduce bugs.

bt-s commented on src/components/Account/Avatar.js:
```diff
@@ -77,11 +87,11 @@ const Avatar = props => {
 
 Avatar.propTypes = {
-  avatarURL: PropTypes.string
+  avatarURL: PropTypes.string,
```
@Envl it looks like your Prettier configuration has changed.
Throughout the code base, I have disabled trailing commas after the last item of a list or hashmap.

bt-s commented on src/components/Authentication/SignUp.js:
```diff
@@ -57,11 +57,23 @@ const SignUpFormBase = props => {
+            photoURL:
+              'https://firebasestorage.googleapis.com/v0/b/bookio.appspot.com/o/images%2Frobot.png?alt=media&token=d0cc8a48-aaad-4e3b-9c08-3e341ed8165e',
+          })
+          .then(function() {
```
Same here, please add some proper error handling.

Envl replied:
Just according to the standard, we should catch the error, but there are no errors now. So I don't know what to catch.

bt-s commented on src/components/Authentication/SignUp.js:
```diff
@@ -57,11 +57,23 @@ const SignUpFormBase = props => {
+          .updateProfile({
+            displayName: form.username,
+            photoURL:
+              'https://firebasestorage.googleapis.com/v0/b/bookio.appspot.com/o/images%2Frobot.png?alt=media&token=d0cc8a48-aaad-4e3b-9c08-3e341ed8165e',
```
Should this URL be hard-coded in here? If so, maybe define it as some sort of const or macro at the top of the file.

Envl replied:
That's good. I just uploaded a default picture for all new users.

bt-s commented on src/components/Firebase/firebase.js:
```diff
@@ -96,11 +96,12 @@ class Firebase {
-              emailVerified: authUser.emailVerified,
-              providerData: authUser.providerData,
-              ...dbUser
+              // uid: authUser.uid,
```
It seems like these comments could be deleted.

bt-s commented on src/components/Session/withAuthentication.js:
```diff
@@ -10,17 +10,17 @@ const withAuthentication = Component => {
       super(props);
 
-      this.props.onSetAuthUser(JSON.parse(localStorage.getItem('authUser')));
+      // this.props.onSetAuthUser(JSON.parse(localStorage.getItem('authUser')));
```
It seems like this comment could be deleted.

bt-s commented on src/components/Session/withAuthentication.js:
```diff
@@ -10,17 +10,17 @@ const withAuthentication = Component => {
       this.listener = this.props.firebase.onAuthUserListener(
         authUser => {
-          localStorage.setItem('authUser', JSON.stringify(authUser));
+          // localStorage.setItem('authUser', JSON.stringify(authUser));
```
Same for this one.

bt-s commented on src/components/Session/withAuthentication.js:
```diff
@@ -10,17 +10,17 @@ const withAuthentication = Component => {
         },
         () => {
-          localStorage.removeItem('authUser');
+          // localStorage.removeItem('authUser');
```
Same for this one.

bt-s reviewed (approved):
Nice catch!

The with... components are HOCs. This is quite idiomatic in React (have a look at withRouter from the react-router-dom package for example). I think that we should generally not change this, however, if you have an alternative that you would like to suggest I would be really interested in hearing it!

Please also have a look at the commits that I have pushed.

Envl committed:
add back a missing bracket

Envl committed:
Merge branch 'link-avatar-image' of https://github.com/bt-s/ipdw-project into link-avatar-image

Envl committed:
delete redundant code
make URL to default portrait a CONST
